### PR TITLE
Users

### PR DESCRIPTION
**Pull request by andrestobelem**
# Summary
* I added the User owner for Articles and Comments.
* Only the owner can edit and delete his own comments and articles.
* I don't know how to secure this on a controller. I did it only in ArticleController::edit with an 'If'. But I think that it should be another way, some kind of assertion.

# Trello Card
https://trello.com/c/JIwACs82

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -18,11 +20,16 @@ def new
   end
 
   def edit
-    article
+    if article.user_id == current_user.id
```
Comment (epintos) [inline app/controllers/articles_controller.rb]: Don't worry about this! you will use pundit for authentication

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -18,11 +20,16 @@ def new
...
+      article
+    else
+      redirect_to @article
+    end
   end
 
   def create
     @article = Article.new(article_params)
+    @article.user_id = current_user.id
```
Comment (epintos) [inline app/controllers/articles_controller.rb]: Just for you to know. `@article.user = current_user` will also work. The ORM knows how to translate this

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -1,6 +1,10 @@
 class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
   def create
-    @comment = article.comments.create(comment_params)
+    @comment = article.comments.create(
```
Comment (epintos) [inline app/controllers/comments_controller.rb]: Thats okay, this is another way of doing the user assignation. The same is valid for the id

File: app/models/article.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 class Article < ActiveRecord::Base
   has_many :comments, dependent: :destroy
+  belongs_to :user
+  validates :user_id, presence: true
```
Comment (epintos) [inline app/models/article.rb]: Here you could use user also

File: app/views/articles/_article.html.slim [hunk 5]
```diff
@@ -7,13 +7,16 @@
...
       footer.article-meta
         span.clock 
           = l article.created_at, format: :long
         span.comments-link 
           = link_to "#{article.comments.count} #{Comment.model_name.human(count: article.comments.count)}", article_path(article)
-        span.article-footer-link
-          = link_to t('actions.edit'), edit_article_path(article)
-        span.article-footer-link
-          = link_to t('actions.delete'), article_path(article), method: :delete, data: { confirm: t('messages.are_you_sure') }
+        - if user_signed_in? && current_user.id == article.user.id
```
Comment (epintos) [inline app/views/articles/_article.html.slim]: Same here you will use pundit

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (andrestobelem) [inline app/controllers/articles_controller.rb]: Yes, I realized later and I changed with the pundit's card.

File: app/views/articles/_article.html.slim (hunk 5 above)
Comment (andrestobelem) [inline app/views/articles/_article.html.slim]: Yes, I did it later.

**Comment by epintos** [comment]
:+1:

**Pull request merged**